### PR TITLE
Add anisotropic filtering to GLES2 backend

### LINKED ISSUE #45641: No anisotropic filtering with GLES2 backend
**Godot version:**

3.2.4.rc1

**OS/device including version:**

Linux Mint 20.1, Radeon HD 7850, amdgpu driver

**Issue description:**

I get no anisotropic filtering at all in GLES2, at least from what I can tell. It works fine in the same project with GLES3.

Also, there is no option to change the anisotropic filtering level in the Project Settings when GLES2 is active. This setting can still be adjusted by switching to GLES3, changing the setting, and switching back, but it has no visible effect in GLES2.

I'm told anisotropic filtering used to work with the GLES2 backend, which is why I'm reporting this as a bug rather than a feature request.

**Steps to reproduce:**
Just run the attached project. Notice how much blurrier the ground is as it approaches the horizon in GLES2 mode, but not in GLES3 mode.

**Minimal reproduction project:**

[anisotropy.zip](https://github.com/godotengine/godot/files/5905955/anisotropy.zip)

### PR DESCRIPTION
**Pull request by furrykef**
Fixes #45641

I made this change by copying the relevant code from the GLES3 backend to the GLES2 backend. In so doing, I noticed a strange line, line 8358 of rasterizer_storage_gles3.cpp:

`config.use_anisotropic_filter = config.extensions.has("rendering/quality/filters/anisotropic_filter_level");`

The strange thing about it is, as far as I can tell, this variable soon gets overwritten without having been used. Still, I copied this line even though I'm not sure it has any effect.

(EDIT: I have discussed this line with other devs and removed it from the GLES2 backend. It's still in the GLES3 backend, though.)

(EDIT: It's gone from both backends now.)

### CODE REVIEW

File: drivers/gles2/rasterizer_gles2.cpp
Comment (clayjohn) [inline drivers/gles2/rasterizer_gles2.cpp:484]: Now that both GLES2 and GLES3 have this setting, it should be defined in visual_server.cpp here:
https://github.com/godotengine/godot/blob/0b6dd929c677baad3888579c3f237db43f19592c/servers/visual_server.cpp#L2439-L2441

You will need to remove it from rasterizer_gles3.cpp as well

Comment (furrykef) [reply drivers/gles2/rasterizer_gles2.cpp]: Yeah, that code duplication bugged me, too. I've made the change, and since we're now modifying the GLES3 code, I decided to remove the extraneous write to config.use_anisotropic_filter in rasterizer_storage_gles3.cpp while I was at it.

**Comment by lawnjelly** [comment]
Looks fine to me although I haven't tested. Great first PR. :+1: 

Seeing the duplication in `texture_set_data` and `texture_set_flags` reminded me I noticed a lot of code duplication here when converting this for Godot 4.0. In there I used a separate function to do all the common stuff between the two functions. It should be fine to leave like this though in 3.2.

**Comment by akien-mga** [comment]
Looks good to me! I'll let @clayjohn do the final review.

As a tip for future PRs, you might want to make them from a dedicated branch instead of your `3.2` branch, so that you can easily create new branches for new PRs even before the first one is merged.

**Comment by Calinou** [comment]
I'd like to do some testing on Android and HTML5 before this is merged, just in case :slightly_smiling_face:

**Review by clayjohn — APPROVED** [review]
Great work!

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**
